source/destructurable.d.ts

```diff
@@ -0,0 +1,47 @@
+import {type Simplify} from './simplify';
+import {type UnionToIntersection} from './union-to-intersection';
+
+/**
+Convert a union type of objects to a destructurable type.
+
```

Comment:
There needs to be a more in-depth explanation of the problem space and how it's solved here, and why the type is useful.